File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/944. Delete Columns to Make Sorted/Python solution/944. Delete Columns to Make Sorted.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        n = len(strs)
        c = len(strs[0])
        delete_count = 0
        all_strs = ''
        for i in range(n):
            all_strs += strs[i]
        for i in range(c):
            a, b = i, i + c
            for j in range(n - 1):
                if all_strs[a] > all_strs[b]:
                    delete_count += 1
                    break
                a += c
                b += c
        
        return delete_count
                

File: A2SV-Community-Education/Div-3-Group-3-2024/LeetCode/Easy/944. Delete Columns to Make Sorted/C++ solution/944. Delete Columns to Make Sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int n = strs.size();
        int c = strs[0].size();
        int delete_count = 0;
        string all_strs = "";
        for (int i = 0; i < n; i++)
            all_strs += strs[i];
        
        for (int i = 0; i < c; i++)
        {
            int a = i, b = i + c;
            for (int j = 0; j < n - 1; j++)
            {
                if (all_strs[a] > all_strs[b])
                {
                    delete_count++;
                    break;
                }
                a += c;
                b += c;
            }
        }
        return delete_count;
    }
};